internal/controller/controllers/agentserviceconfig_controller.go

```diff
@@ -178,11 +174,11 @@ func (r *AgentServiceConfigReconciler) ensureStorage(ctx context.Context, instan
 	if err := r.Client.Get(ctx, types.NamespacedName{Name: pvc.Name, Namespace: pvc.Namespace}, found); err != nil {
 		if apierrors.IsNotFound(err) {
 			r.Log.Info("Creating PVC", "Namespace", pvc.Namespace, "Name", pvc.Name)
-			err := r.Client.Create(ctx, pvc)
-			if err != nil {
-				r.Log.Error(err, "Create PVC failed", "Namespace", pvc.Namespace, "Name", pvc.Name)
+			errCreate := r.Client.Create(ctx, pvc)
```

Comment:
I guess we could just `return r.Client.Create(ctx, pvc)` since we already have handling of non `nil` err in the caller.